jax/scipy/stats/dirichlet.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as onp
+import scipy.stats as osp_stats
+
+from ... import lax
+from ...numpy.lax_numpy import _promote_args_like, _constant_like, _wraps, all, sum
+from ..special import gammaln, xlogy
+
+
+def _is_simplex(x):
+    x_sum = sum(x, axis=-1)
+    return all(x > 0, axis=-1) & (x_sum <= 1) & (x_sum > 1 - 1e-6)
+
+
+@_wraps(osp_stats.dirichlet.logpdf)
+def logpdf(x, alpha):
+    args = (onp.ones((0,), lax.dtype(x)), onp.ones((1,), lax.dtype(alpha)))
+    to_dtype = lax.dtype(osp_stats.dirichlet.logpdf(*args))
```

Comment:
I want to use `_promote_args_like` but scipy dirichlet does not enjoy it (similar to multivariate_normal), so I made a workaround here. I'm not sure if this is a correct way to do because in my system, `osp_stats....logpdf` methods (for all distributions) return float64 result, regardless the inputs are float32. Anyway, the tests pass so I hope that things should work as intended.

Reply:
Sounds reasonable. I had to check for myself why you didn't make both dummy args into size-zero arrays (it's because the latter is used in a min-reduction and so raises an error when the array has size zero).